src/client.rs

```diff
@@ -200,15 +206,17 @@ impl Client {
     /// assert_eq!(result.len(), 1);
     /// assert_eq!(result["foo"], "42");
     /// ```
-    pub fn gets<V: FromMemcacheValueExt>(&self, keys: &[&str]) -> Result<HashMap<String, V>, MemcacheError> {
-        let mut con_keys: HashMap<usize, Vec<&str>> = HashMap::new();
+    pub fn gets<V, K, I>(&self, keys: I) -> Result<HashMap<String, V>, MemcacheError>
+    where
+        V: FromMemcacheValueExt,
+        K: AsRef<str>,
```

Comment:
Keys need not be strings, the server will accept bytes. So, `AsRef<u8>` is more appropriate.

Reply:
The ``check_key_len()`` function requires ``&str``. There might be other places that also require ``&str``. Switching to ``AsRef<u8>`` may be a deeper change than I was planning.